src/Common/FitOnFhir.Common.Tests/RequestHandlerBaseTests.cs

```diff
@@ -0,0 +1,38 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Microsoft.Health.Common.Handler;
+using Microsoft.Health.FitOnFhir.Common.Requests;
+using Xunit;
+
+namespace Microsoft.Health.FitOnFhir.Common.Tests
+{
+    public abstract class RequestHandlerBaseTests<TRequest, TResult>
+        where TRequest : RequestBase
+        where TResult : class
+    {
+        public RequestHandlerBaseTests()
+        {
+        }
+
+        protected IResponsibilityHandler<TRequest, TResult> RequestHandler { get; set; }
+
+        protected TRequest Request { get; set; }
```

Comment:
This property should probably be abstract and get only to force the subclasses to provide a value. Because this value is not set, the test is actually the same as `GivenRequestRouteIsNull_WhenEvaluateCalled_NullReturned` because `Request` is null.

```c#
protected abstract TRequest NonHandledRequest { get; } 
```